tools/rosbag_storage/src/bag.cpp

```diff
@@ -663,8 +895,13 @@ void Bag::appendConnectionRecordToBuffer(Buffer& buf, ConnectionInfo const* conn
 
 void Bag::readConnectionRecord() {
     ros::Header header;
-    if (!readHeader(header))
-        throw BagFormatException("Error reading CONNECTION header");
+    if (!aes_session_key_.empty()) {
+        if (!readEncryptedHeader(header))
```

Comment:
Same comment here. Moving the functionality of the read encrypted header inside the read header method will help.